runtime/protocol/http-client-engines/http-client-engine-ktor/jvm/src/aws/smithy/kotlin/runtime/http/engine/ktor/KtorEngine.kt

```diff
@@ -38,6 +39,7 @@ class KtorEngine(val config: HttpClientEngineConfig) : HttpClientEngineBase("kto
     }
     private val logger = Logger.getLogger<KtorEngine>()
 
+    @OptIn(InternalAPI::class)
```

Comment:
**Question**: Isn't opting into using their internal API a bad thing? Shouldn't we look to rid ourselves of Ktor internal usage instead?

Reply:
After looking at the ktor code for awhile, I'm not seeing a way of doing that.  Given the updated policy on experimental APIs, it would seem that all of `StringValues` becomes experimental (the interface itself references `StringValuesBuilder` which is annotated as Experimental.  Both headers and (query) parameters are based on this type. I'm probably missing something obvious but as it stands I don't see how to construct an instance of Ktor's `HttpRequest` without opting in to internal API usage.

Reply:
After seeing [the KTOR issue](https://youtrack.jetbrains.com/issue/KTOR-3001) pointed out by @aajtodd I agree we have to opt in. Can we please comment these opt-ins with a link to the issue and note that once the issue is resolved, we can remove the opt-ins?

Reply:
Sure, will do.